ibridges/irodsconnector/permission.py

```diff
@@ -1,102 +1,50 @@
-""" permission operations
-"""
-import logging  # noqa: I001
-from typing import Optional
-
+""" permission operations """
+from typing import Iterator
 import irods.access
 import irods.collection
 import irods.exception
-
-from ibridges.irodsconnector.data_operations import DataOperation
-from ibridges.irodsconnector.session import Session
-
+import irods.session
 
 class Permission():
-    """Irods permission operations """
-    _permissions: Optional[dict[str, str]] = None
+    """Irods permission operations"""
 
-    def __init__(self, session: Session):
-        """ iRODS data operations initialization
+    def __init__(self, session, item) -> None:
+        self.session = session
+        self.item = item
 
-            Parameters
-            ----------
-            data_man: dataOperations.DataOperation
-                instance of the Dataoperation class
-            sess_man : Session
-                instance of the Session class
+    def __iter__(self) -> Iterator:
+        for perm in self.session.irods_session.permissions.get(self.item):
+            yield perm
 
-        """
-        self.session = session
+    def __repr__(self) -> str:
```

Comment:
I think this looks a bit more like a `__str__`, no?

Reply:
Yep, changed it into `__str__`.